src/riak_cs_s3_response.erl

```diff
@@ -43,8 +43,11 @@
 -type xmlElement() :: #xmlElement{}.
 
 -type error_reason() :: atom()
-                  | {'riak_connect_failed', term()}
-                  | {'malformed_policy_version', string()}.
+                      | {'riak_connect_failed', term()}
+                      | {'malformed_policy_version', string()}
+                      | {'disk_quota', non_neg_integer(), non_neg_integer()}
+                      | {'bandwidth', non_neg_integer(), non_neg_integer()}
+                      | {'access_count', non_neg_integer(), non_neg_integer()}.
```

Comment:
It seems taking these error reasons into riak cs itself is a little too tightly coupled.
Or, if these three reasons added here is general enough, it may be better to add callback functions for error message, error code and api_error.